### DISCUSSION TITLE
Cant deploy new migration

### DISCUSSION THREAD

**Discussion by carakessler** [prisma/prisma#20487] [Q&A] [answered]
i am attempting to apply a migration which makes a field on a model required (it was previously optional) and deletes a column on that model. since the feature that uses this model isnt live yet, i just manually deleted all entries in the table (callled Project). it is now empty in my postgres DB.
I am attempting to deploy the migration to staging (using railway) and it keeps failing, saying:
```
A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
Migration name: 20230726183513_removebreakmandatoryproj
Database error code: 23502
Database error: ERROR: column "rosterId" of relation "Project" contains null values
DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E23502), message: "column \"rosterId\" of relation \"Project\" contains null values", detail: None, hint: None, position: None, where_: None, schema: Some("public"), table: Some("Project"), column: Some("rosterId"), datatype: None, constraint: None, file: Some("tablecmds.c"), line: Some(5443), routine: Some("ATRewriteTable") }   0: sql_schema_connector::apply_migration::apply_script           with migration_name="20230726183513_removebreakmandatoryproj"             at schema-engine/connectors/sql-schema-connector/src/apply_migration.rs:106   1: schema_core::commands::apply_migrations::Applying migration           with migration_name="20230726183513_removebreakmandatoryproj"             at schema-engine/core/src/commands/apply_migrations.rs:91   2: schema_core::state::ApplyMigrations             at schema-engine/core/src/state.rs:197
```

however, the project table and roster table are both empty. it deployed properly in prod (which has never had anything in the project table) but is failing in staging. not sure what to do here...

this is the migration
```
/*
  Warnings:

  - You are about to drop the column `breakLength` on the `Project` table. All the data in the column will be lost.
  - Made the column `rosterId` on table `Project` required. This step will fail if there are existing NULL values in that column.

*/
-- DropForeignKey
ALTER TABLE "Project" DROP CONSTRAINT "Project_rosterId_fkey";

-- AlterTable
ALTER TABLE "Project" DROP COLUMN "breakLength",
ALTER COLUMN "rosterId" SET NOT NULL;

-- AddForeignKey
ALTER TABLE "Project" ADD CONSTRAINT "Project_rosterId_fkey" FOREIGN KEY ("rosterId") REFERENCES "Roster"("id") ON DELETE RESTRICT ON UPDATE CASCADE;
```
i have wiped the DB clean, and then deployed successfully, then applied the backup and it still is throwing error

**Comment by ludralph** [comment, marked as answer]
Hi @carakessler 👋 

Thank you for raising this question. 

Can you mark the migration as rolled back and redeploy as suggested in this [documentation](https://www.prisma.io/docs/guides/migrate/production-troubleshooting#option-1-mark-the-migration-as-rolled-back-and-re-deploy) or you might want to manually complete the migration steps and resolve as applied. See this [section](https://www.prisma.io/docs/guides/migrate/production-troubleshooting#option-2-manually-complete-migration-and-resolve-as-applied) of the documentation.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by carakessler** [comment]
Yes, the issue was with how i was marking it as resolved on railway. Figured it out! Thanks :)

**Reply by ludralph** [reply]
I'm glad you figured this out 🙌
